test/docfx.Test/lib/MonikerRangeParserTest.cs

```diff
@@ -104,30 +103,78 @@ public void InvalidMonikerRange(string rangeString, string errorMessage)
         [Fact]
         public void TestDuplicateMonikerNameShouldFail()
         {
-            Moniker[] monikers =
+            var monikerDefinition = new MonikerDefinitionModel
             {
-                new Moniker
+                Monikers =
                 {
-                    MonikerName = "netcore-1.0",
-                    Order = 1,
-                    ProductName = ".NET Core",
-                },
-                new Moniker
-                {
-                    MonikerName = "netcore-1.0",
-                    Order = 2,
-                    ProductName = ".NET Core",
-                },
-               new Moniker
-                {
-                    MonikerName = "netcore-2.0",
-                    Order = 2,
-                    ProductName = ".NET Core",
-                },
+                    new Moniker
+                    {
+                        Name = "netcore-1.0",
+                        Product = ".NET Core",
+                    },
+                    new Moniker
+                    {
+                        Name = "netcore-1.0",
+                        Product = ".NET Core",
+                    },
+                   new Moniker
+                    {
+                        Name = "netcore-2.0",
+                        Product = ".NET Core",
+                    },
+                }
             };
-            var exception = Assert.Throws<DocfxException>(() => new MonikerRangeParser(monikers));
+
+            var exception = Assert.Throws<DocfxException>(() => new MonikerRangeParser(monikerDefinition));
             Assert.Equal("moniker-name-conflict", exception.Error.Code);
             Assert.Equal("Two or more moniker definitions have the same monikerName `netcore-1.0`", exception.Error.Message);
         }
+
+        [Fact]
+        public void TestEmptyMonikerNameShouldFail()
+        {
+            var path = $"moniker-definition-test/{Guid.NewGuid()}";
+            
+            File.WriteAllText(path, @"{
+    ""monikers"": [
+        {
+            ""name"": """",
+            ""product"": ""product-test""
+        }
+    ]
+}");
+
+            var exception = Assert.Throws<DocfxException>(() => MonikerRangeParser.Create(path));
+            Assert.Equal("invalid-moniker-definition", exception.Error.Code);
+            Assert.Equal("Invalid moniker definition file: Moniker name cannot be null or empty", exception.Error.Message);
+        }
+
+        [Fact]
+        public void TestEmptyMonikerProductShouldFail()
```

Comment:
>TestEmptyMonikerProductShouldFail [](start = 20, length = 33)

how about consolidating this and the one above using `[Theory]` and `[InlineData]`?